Replicator/IncomingRev.cc

```diff
@@ -237,6 +296,21 @@ namespace litecore { namespace repl {
             }
         }
 
+        // Save the attachments in _attachments
+        std::optional<set<string>> attachmentsFromSG;
+        Value legacyAttachments = root[C4Blob::kLegacyAttachmentsProperty];
+        if (Dict attachments = legacyAttachments.asDict(); attachments) {
+            attachmentsFromSG.emplace();
+            for (Dict::iterator it(attachments); it; ++it) {
+                if (Dict v = it.value().asDict(); v) {
+                    auto digest = v[C4Blob::kDigestProperty];
+                    if (digest.asString()) {
+                        attachmentsFromSG->emplace(digest.asString());
+                    }
+                }
+            }
+        }
+
```

Comment:
For the case of using delta sync and that the customer only deletes an attachment in _attachments but not its corresponding blob in the body, will `_db->applyDelta(_rev->docID, _rev->deltaSrcRevID, jsonBody);` synthesizes _attachments correctly by not adding the attachment back to the _attachments?

Reply:
For the case of using **delta sync** and when the customer only deleted an attachment in _attachments but not its corresponding blob in the body.

Will `C4Document::encodeStrippingOldMetaProperties(root, sk)` in line 320 add the attachment back in the _attachments?

Reply:
I think it will not otherwise the test will not pass but I'm asksing to make sure that we are synthezing _attachments correctly.

Reply:
Apply delta will not put the blob back to _attachments. However, if we save it to db, we won't save _attachments, and yes, next time as we synthesize _attachments, it will be put back.That's why I said if we let it succeed, next time push will push the attachment to SG

Reply:
It goes like, we synthesize the base of the delta, namely the current revision, and get _attachments before the delta. If the current revision is in the sound state, one of the attachment will be removed after applying the delta.